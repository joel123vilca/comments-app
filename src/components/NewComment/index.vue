<template>
  <div class="new__comment">
    <div class="title">
      <h4>Tu comentario</h4>
    </div>
    <input type="text" v-model="comment" />
    <div class="actions">
      <button class="cancel" @click="cancelComment">Cancelar</button>
      <button class="save" @click="newComment">Guardar</button>
    </div>
  </div>
</template>
<script src="./newComment.js"></script>
<style src="./newComment.scss" lang="scss"></style>
