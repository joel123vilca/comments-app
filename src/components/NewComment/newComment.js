import { mapActions } from "vuex";
export default {
  data() {
    return {
      comment: ""
    };
  },
  methods: {
    ...mapActions({
      saveComment: "SAVE_COMMENT"
    }),
    newComment() {
      this.saveComment({ comment: this.comment });
      this.comment = "";
    },
    cancelComment() {
      this.comment = "";
    }
  }
};
