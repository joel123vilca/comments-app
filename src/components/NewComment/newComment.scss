.new__comment {
  margin-top: 10px;
  width: 100%;
  background: white;
  -webkit-box-shadow: 0 8px 6px -6px rgba(99, 99, 99, 0.2);
  -moz-box-shadow: 8px 8px 6px -6px rgba(99, 99, 99, 0.2);
  box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
  border-radius: 10px;
  .title {
    display: flex;
    margin-left: 20px;
  }
  input {
    width: 400px;
    height: 40px;
    display: flex;
    margin-left: 20px;
  }
  .actions {
    display: flex;
    margin: 10px;
    padding-bottom: 20px;
    .cancel {
      width: 100px;
      height: 30px;
      color: white;
      background: #152b72;
      border: 1px solid #152b72;
      border-radius: 10px;
      margin-left: 20px;
      cursor: pointer;
    }
    .save {
      width: 100px;
      height: 30px;
      color: white;
      background: #ff5733;
      border: 1px solid #ff5733;
      border-radius: 10px;
      margin-left: 20px;
      cursor: pointer;
    }
  }
}
