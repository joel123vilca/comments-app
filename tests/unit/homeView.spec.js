import { shallowMount } from "@vue/test-utils";
import NewComment from "@/components/NewComment";

describe("comment view", () => {
  const build = () => {
    const wrapper = shallowMount(NewComment);

    return {
      wrapper,
      input: () => wrapper.find("input"),
      button: () => wrapper.find("button")
    };
  };

  it("renders the component", () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("renders the right", () => {
    const { input, button } = build();
    expect(input().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });
});
