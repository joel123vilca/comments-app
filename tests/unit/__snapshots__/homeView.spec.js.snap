// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comment view renders the component 1`] = `
<div class="new__comment">
  <div class="title">
    <h4>Tu comentario</h4>
  </div> <input type="text">
  <div class="actions"><button class="cancel">Cancelar</button> <button class="save">Guardar</button></div>
</div>
`;
